src/main/java/net/minecraftforge/client/ForgeHooksClient.java

```diff
@@ -820,6 +824,19 @@ public static void loadLayerDefinitions(ImmutableMap.Builder<ModelLayerLocation,
         layerDefinitions.forEach((k, v) -> builder.put(k, v.get()));
     }
 
+    private static final Map<SkullBlock.Type, Function<EntityModelSet,SkullModelBase>> skullModelConstructors = new HashMap<>();
+
+    /** Call during {@link EntityRenderersEvent.RegisterLayerDefinitions} via the event method */
+    public static void registerSkullModel(SkullBlock.Type type, Function<EntityModelSet,SkullModelBase> constructor)
+    {
+        skullModelConstructors.put(type, constructor);
+    }
```

Comment:
Please move this method to the event, so that it is impossible use it outside of the event.
If necessary, make a new event.

Reply:
In what way do you suggest moving it into the event? Should the static map be a field in the event? Not sure of another good way to grant mutable access to a private variable from another class

Reply:
Yes, the idea is that we limit exactly what can mutate the map and when.
Passing the maps as constructor arguments to the event is the simplest way. May need some rejiggering to where the event is constructed tho.
Main reason is we want to be explicitly clear when these things should be registered, by making it impossible {by non hacky means} to register it at another time. Instead of relying on modders reading the comment.